internal/exec/config_fetcher.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"net/http"
+	"net/url"
+
+	"github.com/coreos/ignition/v2/config"
+	"github.com/coreos/ignition/v2/config/shared/errors"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/ignition/v2/internal/state"
+	"github.com/coreos/ignition/v2/internal/util"
+
+	latest "github.com/coreos/ignition/v2/config/v3_4_experimental"
+	"github.com/coreos/ignition/v2/config/v3_4_experimental/types"
+)
+
+type ConfigFetcher struct {
+	Logger  *log.Logger
+	Fetcher *resource.Fetcher
+	State   *state.State
+}
+
+// RenderConfig evaluates "ignition.config.replace" and "ignition.config.append"
+// in the given config and returns the result. If "ignition.config.replace" is
+// set, the referenced and evaluted config will be returned. Otherwise, if
+// "ignition.config.append" is set, each of the referenced configs will be
+// evaluated and appended to the provided config. If neither option is set, the
+// provided config will be returned unmodified. An updated fetcher will be
+// returned with any new timeouts set.
```

Comment:
Might as well update this comment block for spec 3 config merging while we're here.